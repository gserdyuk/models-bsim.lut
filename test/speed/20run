#!/bin/bash

echo "20 run - timing_test" > 20run.out

./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out

./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out

./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out

./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out
./timing_test >> 20run.out

echo "20 run - timing_test.lut" >> 20run.out

./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out

./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out

./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out

./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out
./timing_test.lut >> 20run.out

cat 20run.out |head -n 21 |tail -n 20 |gawk ' {a+=$1;} END {print ("average norm",a/20);} ' >>20run.out
cat 20run.out             |tail -n 20 |gawk ' {a+=$1;} END {print ("average lut" ,a/20);} ' >>20run.out

cat 20run.out| tail -n 2| gawk ' /norm/ {an=$3;} /lut/ {al=$3;} END {print "lut/norm=",al/an;}' >>20run.out

cat 20run.out |tail -n 1
echo OK
