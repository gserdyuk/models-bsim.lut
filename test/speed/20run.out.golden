20 run - timing_test
 5.98
 5.58
 5.53
 5.49
 5.55
 5.66
 5.51
 5.52
 5.56
 5.52
 5.55
 5.56
 5.49
 5.53
 5.56
 5.56
 5.52
 5.48
 5.52
 5.68
20 run - timing_test.lut
 3.46
 3.47
 3.55
 3.42
 3.51
 3.55
 3.48
 3.42
 3.7
 3.72
 3.5
 3.83
 3.43
 3.43
 3.4
 3.46
 3.44
 3.57
 3.73
 3.48
average norm 5.5675
average lut 3.3545
lut/norm= 0.602515
